torch/csrc/distributed/c10d/socket.cpp

```diff
@@ -193,6 +194,43 @@ class SocketImpl {
   Handle hnd_;
   const std::optional<std::string> remote_;
 };
+
+std::string formatSockAddr(const struct ::sockaddr* addr, socklen_t len) {
+  char host[NI_MAXHOST], port[NI_MAXSERV]; // NOLINT
+
+  if (int err = ::getnameinfo(
+          addr, len, host, NI_MAXHOST, port, NI_MAXSERV, NI_NUMERICSERV)) {
+    C10D_WARNING(
+        "The hostname of the client socket cannot be retrieved. err={}", err);
+
+    // if we can't resolve the hostname, display the IP address
+    if (addr->sa_family == AF_INET) {
+      struct sockaddr_in* psai = (struct sockaddr_in*)&addr;
+      char ip[INET_ADDRSTRLEN];
+      if (inet_ntop(addr->sa_family, &(psai->sin_addr), ip, INET_ADDRSTRLEN) !=
+          NULL) {
+        return fmt::format("{}:{}", ip, psai->sin_port);
+      }
+    } else if (addr->sa_family == AF_INET6) {
+      struct sockaddr_in6* psai = (struct sockaddr_in6*)&addr;
+      char ip[INET6_ADDRSTRLEN];
+      if (inet_ntop(
+              addr->sa_family, &(psai->sin6_addr), ip, INET6_ADDRSTRLEN) !=
+          NULL) {
+        return fmt::format("[{}]:{}", ip, psai->sin6_port);
+      }
+    }
+
+    C10_THROW_ERROR(
```

Comment:
Why throw here? Will this kill the program?

Reply:
This was already there and not causing issues so I think it's fine to leave. This case would be pretty weird to hit -- only happening if there's a more fundamental bug in the code (i.e. passing an uninitialized address or a non ipv4/v6 address). I don't think we've ever hit this so I'd prefer to just leave it in since it might catch issues in the tests during refactoring, etc